pkg/alertmanager/operator.go

```diff
@@ -763,13 +763,16 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return errors.Wrap(err, "failed to retrieve statefulset")
 	}
 
-	oldSpec := appsv1.StatefulSetSpec{}
+	oldSs := &appsv1.StatefulSet{}
```

Comment:
(nit) existing rather than old?
```suggestion
	existingStatefulSet := &appsv1.StatefulSet{}
```